core/src/test/scala/org/broadinstitute/dsde/rawls/deltalayer/GcsDeltaLayerWriterSpec.scala

```diff
@@ -0,0 +1,118 @@
+package org.broadinstitute.dsde.rawls.deltalayer
+
+import org.broadinstitute.dsde.rawls.TestExecutionContext
+import org.broadinstitute.dsde.workbench.model.google.GcsBucketName
+import org.scalatest.flatspec.AnyFlatSpec
+import TestExecutionContext.testExecutionContext
+import akka.actor.ActorSystem
+import com.google.cloud.storage.Storage.BlobWriteOption
+import com.google.cloud.storage.contrib.nio.testing.LocalStorageHelper
+import com.google.cloud.storage.{BlobInfo, Storage, StorageException}
+import org.broadinstitute.dsde.rawls.model.AttributeUpdateOperations.{AddUpdateAttribute, AttributeUpdateOperation, EntityUpdateDefinition}
+import org.broadinstitute.dsde.rawls.model.{AttributeName, AttributeString, DeltaInsert, Destination, RawlsUserSubjectId}
+import org.broadinstitute.dsde.workbench.google2.GcsBlobName
+import org.joda.time.DateTime
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.RecoverMethods.recoverToExceptionIf
+import org.scalatest.concurrent.Eventually
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatestplus.mockito.MockitoSugar.mock
+
+import java.util.UUID
+import scala.util.Try
+
+class GcsDeltaLayerWriterSpec extends AnyFlatSpec with GcsStorageTestSupport with Eventually with Matchers {
+
+  implicit val actorSystem: ActorSystem = ActorSystem("GcsDeltaLayerWriterSpec")
+
+  behavior of "GcsDeltaLayerWriter"
+
+  it should "write the specified file to the bucket" in {
+    val bucket = GcsBucketName("successful-write")
+
+    // explicitly create the storage so we can query it directly
+    val storage = LocalStorageHelper.getOptions.getService
+    val deltaLayerWriter = getGcsWriter(bucket, Some(storage))
+
+    // create the object to be written
+    val dest = Destination(UUID.randomUUID(), UUID.randomUUID())
+    val attrUpdates: Seq[AttributeUpdateOperation] = Seq(AddUpdateAttribute(AttributeName.withDefaultNS("attrName"), AttributeString("attrValue")))
+    val entityUpdates: Seq[EntityUpdateDefinition] = Seq(EntityUpdateDefinition("name", "type", attrUpdates))
+    val testInsert = DeltaInsert("vTest", UUID.randomUUID(), new DateTime(), RawlsUserSubjectId("1234"), dest, entityUpdates)
+
+    // calculate expected file path and contents
+    val expectedPath = deltaLayerWriter.filePath(testInsert)
+    val expectedContents = deltaLayerWriter.serializeFile(testInsert)
+
+    // write the object via Delta Layer
+    deltaLayerWriter.writeFile(testInsert) map { _ =>
+      // confirm we wrote the file correctly. The LocalStorageHelper in-memory storage engine
+      // seems to have some delays, so use eventually to allow for delays and retries
+      eventually (timeout(Span(30, Seconds)), interval(Span(500, Millis))) {
```

Comment:
Is there any guidance (or any general info) for how long this usually takes? Wondering if there's room for this step to potentially become flaky as-is.

Reply:
@ungwudik I am not aware of any info on this. I am basing my changes on what I saw empirically from running the tests a bunch of times. When I ran them I saw delays of less than 5 seconds (usually < 2), so I thought a 30-second timeout would be ok.